zstacklib/zstacklib/utils/bash.py

```
@@ -51,15 +51,20 @@ def bash_eval(raw_str, ctx=None):
 
     cmd = bash_eval(cmd, ctx)
     p = shell.get_process("/bin/bash", pipe=True)
     if pipe_fail:
         cmd = 'set -o pipefail; %s' % cmd
-    o, e = p.communicate(cmd.encode())
+    try:
+        o, e = p.communicate(cmd.encode(), timeout=timeout)
+    except subprocess.TimeoutExpired:
+        p.kill()
+        p.communicate()
+        raise BashError('bash command timed out after %ss: %s' % (timeout, cmd))
```

Ruff (0.15.7)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)